GoogleForCreators/web-stories-wp
# Code Quality: Refactor `setOrCreateImage` function

## Context



## Summary

Refactor logic of setOrCreateImage. Instead of creating an image tag and append it to body tag, just pass a image tag into the color thief. Remove lots of unneeded logic and generally make it easier to read. 

## Relevant Technical Choices



## To-do



## User-facing changes

<img width="1896" alt="Screenshot 2022-01-10 at 18 02 00" src="https://user-images.githubusercontent.com/237508/148824704-2f9ed4ec-d83c-48e6-9a8c-b4c8e087fedb.png">


## Testing Instructions






This PR can be tested by following these steps:

1. Create a story. 
2. Set the background of first page to a image of a solid color, like white. 
3. Create text element.
4. Add some text
5. Change text color the same color as background.
6. Publish story. 
7. Go to check. See accessibility warning. 

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10025
